scripts/Editor/DllManipulatorEditor.cs

```diff
@@ -93,12 +106,67 @@ public override void OnInspectorGUI()
             EditorGUILayout.HelpBox(INFO_BOX_GUI_CONTENT, MessageType.Info);
 
             DrawOptions(t.Options);
+
+            DetectOptionChanges(t);
+            
             EditorGUILayout.Space();
 
+            DrawCurrentState(t);
+        }
+
+        /// <summary>
+        /// Detects whether the <see cref="DllManipulatorScript.Options"/> have changed, both relative to the previous
+        /// options and the <see cref="DllManipulator.Options"/> if we are currently initialized.
+        /// </summary>
+        /// <param name="t">The OnInspectorGUI target</param>
+        private void DetectOptionChanges(DllManipulatorScript t)
+        {
+            // Set the target as dirty so changes can be saved, if there are changes
+            if (GUI.changed)
+            {
+                if (!t.Options.Equals(_prevOptions))
+                {
+                    // If the options have changed then update the _prevOptions and notify there are changes to be saved
+                    // CloneTo is used to ensure a deep copy is made
+                    t.Options.CloneTo(_prevOptions);
+                    EditorUtility.SetDirty(target);
+                }
+            }
+
+            // Allow Reinitializing DllManipulator if there are changes
+            if (DllManipulator.Options != null && !t.Options.Equals(DllManipulator.Options))
+            {
+                if (DllManipulator.Options.loadingMode == DllLoadingMode.Preload)
+                {
+                    if (GUILayout.Button(REINITIALIZE_WITH_CHANGES_PRELOADED_GUI_CONTENT))
+                        t.Reinitialize();
+                }
+                else if(GUILayout.Button(REINITIALIZE_WITH_CHANGES_LAZY_GUI_CONTENT))
+                    t.Reinitialize();
+            }
+            
+            // When enabling enableInEditMode for the first time, allow immediately initializing without waiting for OnEnable
+            if(DllManipulator.Options == null && t.Options.enableInEditMode && !EditorApplication.isPlaying && 
+               GUILayout.Button(INITIALIZE_ENABLED_EDIT_MODE_GUI_CONTENT))
```

Comment:
I would maybe move that button to be close to the 'enable in edit mode' option

Reply:
I mean you enable it and this shows up right next to it so that you know you have to click it first for it to work, sth like warning button i guess

Reply:
I think its reasonably clear when the button pops up. If the user immediately clicks the initialize button and then changes the options below they will have to unnecessarily reinitialize. Having the button at the bottom implies that the other options also affect its behaviour.

Also in the code it is clearer as the functionality about re/initializing is together in `DetectOptionChanges`.

Reply:
Yup

Reply:
Although, it looks like `Options` in `DllManipulator` can now be private?

Reply:
We use it in the Script and Editor to check if the DllManip is initialized/there are changes to be applied. We could make the setter private?

Reply:
yes